integration-cli/docker_cli_swarm_test.go

```diff
@@ -1888,3 +1888,72 @@ func (s *DockerSwarmSuite) TestSwarmServiceLsFilterMode(c *check.C) {
 	c.Assert(out, checker.Contains, "top1")
 	c.Assert(out, checker.Not(checker.Contains), "top2")
 }
+
+func (s *DockerSwarmSuite) TestServiceCreateWithDuplicateNetworkNames(c *check.C) {
+	d := s.AddDaemon(c, true, true)
+
+	name := "foo"
+	networkCreateRequest := types.NetworkCreateRequest{
+		Name: name,
+		NetworkCreate: types.NetworkCreate{
+			CheckDuplicate: false,
+			Driver:         "bridge",
+		},
+	}
+
+	// Create networks with the same name, 2 in local scope and 1 in swarm scope
+	var n1 types.NetworkCreateResponse
+	status, body, err := d.SockRequest("POST", "/networks/create", networkCreateRequest)
+	c.Assert(err, checker.IsNil, check.Commentf(string(body)))
+	c.Assert(status, checker.Equals, http.StatusCreated, check.Commentf(string(body)))
+	c.Assert(json.Unmarshal(body, &n1), checker.IsNil)
+
+	var n2 types.NetworkCreateResponse
+	status, body, err = d.SockRequest("POST", "/networks/create", networkCreateRequest)
+	c.Assert(err, checker.IsNil, check.Commentf(string(body)))
+	c.Assert(status, checker.Equals, http.StatusCreated, check.Commentf(string(body)))
+	c.Assert(json.Unmarshal(body, &n2), checker.IsNil)
+
+	var n3 types.NetworkCreateResponse
+	// Dupliates with name but with different driver
+	networkCreateRequest.NetworkCreate.Driver = "overlay"
+	status, body, err = d.SockRequest("POST", "/networks/create", networkCreateRequest)
+	c.Assert(err, checker.IsNil, check.Commentf(string(body)))
+	c.Assert(status, checker.Equals, http.StatusCreated, check.Commentf(string(body)))
+	c.Assert(json.Unmarshal(body, &n3), checker.IsNil)
+
+	// Create Service with the same name
+	out, err := d.Cmd("service", "create", "--name", "top", "--network", name, "busybox", "top")
+	c.Assert(err, checker.IsNil, check.Commentf(out))
```

Comment:
Using this will give you better error messages when the assertion fails:

```go
result := icmd.RunCmd(d.Command(...))
result.Assert(c, icmd.Success)
```

Same with all the calls below. You can also check for stdout in the same call with:

```go
result.Assert(c, icmd.Expected{
    Out: n3.ID,
})
```

And you can get the stdout from `result.Stdout()`, and stderr from `result.Stderr()`